.github/workflows/kubeflex-compatibility.yml

```diff
@@ -0,0 +1,134 @@
+name: KubeFlex Compatibility Matrix
+
+on:
+  schedule:
+    - cron: '0 4 * * *' # Runs daily at 4 AM UTC
+  workflow_dispatch:
+    inputs:
+      kubestellar_ref:
+        description: 'KubeStellar git ref (branch, tag, or commit)'
+        required: true
+        default: 'main'
+
+permissions:
+  contents: read
+
+jobs:
+  setup-and-deploy:
+    name: Setup Environment and Deploy KubeStellar
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+        with:
+          ref: ${{ github.event.inputs.kubestellar_ref || 'main' }}
+
+      - uses: actions/setup-go@d35c59abb061a4a6fb18e82ac0862c26744d6ab5
+        with:
+          go-version: '1.24'
+          cache: true
+
+      - name: Install kubectl
+        uses: azure/setup-kubectl@776406bce94f63e41d621b960d78ee25c8b76ede
+
+      - uses: ko-build/setup-ko@d006021bd0c28d1ce33a07e7943d48b079944c8d
+
+      - name: Install common dependencies
+        run: |
+          bash <(curl -L https://raw.githubusercontent.com/open-cluster-management-io/clusteradm/main/install.sh)
+          go install github.com/onsi/ginkgo/v2/ginkgo
+
+      - name: Setup KubeStellar
+        run: |
+          cd test/e2e/
+          ./common/setup-kubestellar.sh
+
+      - name: Upload kubeconfig
+        uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02
+        with:
+          name: kubeconfig
+          path: ~/.kube/config
+
+  generate-matrix:
+    name: Generate KubeFlex version matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.generate.outputs.matrix }}
+    steps:
+      - name: Generate kflex version matrix
+        id: generate
+        run: |
+          MIN_VERSION="v0.8.0"
+          VERSIONS=$(curl -s "https://api.github.com/repos/kubestellar/kubeflex/releases" | jq -r '.[].tag_name' | grep -E "^v[0-9]+\.[0-9]+\.[0-9]+$" | sort -V)
+
+          MATRIX_VERSIONS='[]'
+          while read -r v; do
+            if [[ "$(printf '%s\n%s' "$v" "$MIN_VERSION" | sort -V | head -n1)" == "$MIN_VERSION" ]]; then
```

Comment:
It seems to me that the test here is more complicated than necessary. Wouldn't either of the following suffice?

- `! [[ "$v" < "$MIN_VERSION" ]]`
- `[[ "$v" == "$MIN_VERSION" ]] || [[ "$v" > "$MIN_VERSION" ]]`